pwnlib/tubes/process.py

```diff
@@ -631,25 +651,26 @@ def poll(self, block = False):
 
         # In order to facilitate retrieving core files, force an update
         # to the current working directory
-        _ = self.cwd
+        if sys.platform.startswith("linux"):
```

Comment:
I don't normally get nit-picky, but you just changed a bunch of lines you didn't have to.

Instead of `if linux` and then indenting the code, you should have `if not linux` and return early.

All of that said, and as I've pointed out above, valid target OSes are linux, android, freebsd, netbsd, and many others.

You can't just check for "linux".

If you look at my recommendations above, this could just be:

```
if context.windows:
  return foobar
```